### ISSUE TITLE
[Feature Request] Option to get hosts file from http url

### ISSUE THREAD

**Issue by adb76** [0xERR0R/blocky#867]
Hello,

as far a I understood the configuration options correctly, currently it is only possible to get a hosts configuration from a file on the file system:

```
hostsFile:
  filePath: /etc/hosts
  hostsTTL: 24h
  refreshPeriod: 6h
```

It would be great to have an option to get it additionally or optionally also from an http url (analog to the ad blacklists), e.g.:

```
hostsFile:
  hosts:
    - /etc/hosts
    - https://some.url/hosts
  hostsTTL: 24h
  refreshPeriod: 6h
```

What is the idea behind it? Well, I am using an DNS service to avoid geoip blocking, which offers an URL to an up to date hosts list, which of course can change over time. So the idea is to avoid manual download of that list - or to avoid to forget it to do so ;-) 

Kind regards.

**Comment by ThinkChaos** [comment]
Started working on this by unifying the `HostsFileResolver` and `ListCache` download and parsing code.  

From a quick look at go-getter, I'm not sure it'd be a quick replacement since it doesn't seem to have a way to set the HTTP transport (we need it so the resolving uses the bootstrap DNS). Also we'd need to also use the bootstrap for DNS with the other protocols, so that would need a rework anyways.

I'll do a MR with just re-using our existing code, and we can always decide later for go-getter, or something else, to support more protocols if we want.

**Comment by kwitsch** [comment]
@ThinkChaos should this issue be closed since your branch is already merged?

**Comment by ThinkChaos** [comment]
First MR only allows using a link, I'm still working on the multiple sources code.  
Currently setting up a module with sync algorithms and primitives to use. It's mostly done, have it coded and integrated in blocky, working on the CI ATM. Doing a setup to have coverage in PRs with just GH actions, hoping we can reuse that here too since codecov has some quirks.

TL;DR: WIP, coming soon™

**Comment by kwitsch** [comment]
Ok 👍

Was just asking since a successful merge is linked but the issue is still open. 😅
